### ISSUE TITLE
`prefer_const_constructors`: false positive in tenrary expression

### ISSUE THREAD

**Issue by Pante** [dart-lang/linter#4531]
**Describe the issue**
When trying to create a const object in a ternary expression, both parts must either be const or non-const. Having a mix will result in a compilation error.

**To Reproduce**
```dart
class A<T> {
  const A();
}

class B<T> extends A<T> {
  B();
}

```

```dart
A<T> foo<T>(bool flag) => flag ? A() : B(); // warns that A() should be const

A<T> foo<T>(bool flag) => flag ? const A() : B(); // compile error: A value of type 'Object' can't be returned from the function 'foo' because it has a return type of 'A<T>'
```

**Expected behavior**
No warning

**Additional context**
I'm unsure if this is related to #3802.

**Comment by lrhn** [comment]
Adding the `const` changes the type of the literal from `A<T>` to `A<Never>`.
That itself suggests that the lint might be out of place.

The error then comes from `A<Never>` and `B<T>` not having any shared superintetface below `Object`.

**Comment by srawlins** [comment]
Fixed with https://github.com/dart-lang/sdk/commit/f15bcc92707c8e0b8069d24dce4f9b9cb51420f1
